### PR TITLE
[Fix-7196] Remedy a few issues with the docker deployment and the initializing database.

### LINKED ISSUE #7088: [BUG][DOCKER] docker compose api server failed
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/dolphinscheduler/issues?q=is%3Aissue) and found no similar issues.

### What happened

OS: Centos 7 kernel 3.10.0
memory: 8G
Followed the instruction at [https://dolphinscheduler.apache.org/en-us/docs/latest/user_doc/guide/installation/docker.html](https://dolphinscheduler.apache.org/en-us/docs/latest/user_doc/guide/installation/docker.html) 
with Docker 20.10.7 and Docker Compose 2.2.1
Success in starting PostgreSQL, Zookeeper, the master and the worker server, failed to start the API server (keep restarting). Logs show that the connection to 127.0.0.1:5432 was refused.

### What you expected to happen

Failed to connect the api at 127.0.0.1:12345/dolphinscheduler.

### How to reproduce

Follow the instruction at [https://dolphinscheduler.apache.org/en-us/docs/latest/user_doc/guide/installation/docker.html](https://dolphinscheduler.apache.org/en-us/docs/latest/user_doc/guide/installation/docker.html) 

### Anything else

_No response_

### Version

2.0.0

### Are you willing to submit PR?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### LINKED ISSUE #7196: [Bug] [DOCKER] apache/dolphinscheduler:2.0.0 image , column "state" not exist in t_ds_user
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/dolphinscheduler/issues?q=is%3Aissue) and found no similar issues.

### What happened

[INFO] 2021-12-05 18:42:23.832 org.apache.dolphinscheduler.api.aspect.AccessLogAspect:[76] - REQUEST TRANCE_ID:01aadf72-c476-4cae-9ebb-4decaaa17686, LOGIN_USER:NOT LOGIN, URI:/dolphinscheduler/login, METHOD:POST, HANDLER:org.apache.dolphinscheduler.api.controller.LoginController.login, ARGS:{userName=admin}
[ERROR] 2021-12-05 18:42:23.838 org.apache.dolphinscheduler.api.exceptions.ApiExceptionHandler:[46] - 用户登录失败
org.springframework.jdbc.BadSqlGrammarException: 
### Error querying database.  Cause: org.postgresql.util.PSQLException: ERROR: column "state" does not exist
  Position: 124
### Cause: org.postgresql.util.PSQLException: ERROR: column "state" does not exist
  Position: 124
; bad SQL grammar []; nested exception is org.postgresql.util.PSQLException: ERROR: column "state" does not exist
  Position: 124
        at org.springframework.jdbc.support.SQLStateSQLExceptionTranslator.doTranslate(SQLStateSQLExceptionTranslator.java:101)
        at org.springframework.jdbc.support.AbstractFallbackSQLExceptionTranslator.translate(AbstractFallbackSQLExceptionTranslator.java:72)
        at org.springframework.jdbc.support.AbstractFallbackSQLExceptionTranslator.translate(AbstractFallbackSQLExceptionTranslator.java:81)
        at org.springframework.jdbc.support.AbstractFallbackSQLExceptionTranslator.translate(AbstractFallbackSQLExceptionTranslator.java:81)
        at org.mybatis.spring.MyBatisExceptionTranslator.translateExceptionIfPossible(MyBatisExceptionTranslator.java:74)
        at org.mybatis.spring.SqlSessionTemplate$SqlSessionInterceptor.invoke(SqlSessionTemplate.java:440)
        at com.sun.proxy.$Proxy101.selectOne(Unknown Source)
        at org.mybatis.spring.SqlSessionTemplate.selectOne(SqlSessionTemplate.java:159)
        at com.baomidou.mybatisplus.core.override.MybatisMapperMethod.execute(MybatisMapperMethod.java:89)
        at com.baomidou.mybatisplus.core.override.MybatisMapperProxy.invoke(MybatisMapperProxy.java:61)
        at com.sun.proxy.$Proxy106.queryUserByNamePassword(Unknown Source)
        at org.apache.dolphinscheduler.api.service.impl.UsersServiceImpl.queryUser(UsersServiceImpl.java:287)
        at org.apache.dolphinscheduler.api.service.impl.UsersServiceImpl$$FastClassBySpringCGLIB$$ac00af33.invoke(<generated>)
        at org.springframework.cglib.proxy.MethodProxy.invoke(MethodProxy.java:218)
        at org.springframework.aop.framework.CglibAopProxy$DynamicAdvisedInterceptor.intercept(CglibAopProxy.java:687)
        at org.apache.dolphinscheduler.api.service.impl.UsersServiceImpl$$EnhancerBySpringCGLIB$$1302d40b.queryUser(<generated>)
        at org.apache.dolphinscheduler.api.security.impl.pwd.PasswordAuthenticator.login(PasswordAuthenticator.java:32)
        at org.apache.dolphinscheduler.api.security.impl.AbstractAuthenticator.authenticate(AbstractAuthenticator.java:60)
        at org.apache.dolphinscheduler.api.controller.LoginController.login(LoginController.java:104)
        at org.apache.dolphinscheduler.api.controller.LoginController$$FastClassBySpringCGLIB$$a36f5ae2.invoke(<generated>)
        at org.springframework.cglib.proxy.MethodProxy.invoke(MethodProxy.java:218)
        at org.springframework.aop.framework.CglibAopProxy$CglibMethodInvocation.invokeJoinpo

### What you expected to happen

login system normal.

### How to reproduce

use dolphinscheduler:2.0.0 docker image, and exec docker-compose up -d.

其实应该先解决掉连接pg库127.0.0.1连不上，那个问题

### Anything else

_No response_

### Version

2.0.0

### Are you willing to submit PR?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by calvinjiang**
## Purpose of the pull request

This PR will close #7196.
This PR will close #7088.

## Brief change log
The process of building dolphinscheduler image missed the file of datasource.properties.tpl. And fix a few issues in the startup-init-conf.sh. So need to rebuild dolphinscheduler image , and deploy it in the docker.  

I fixed a few issues with the initializing database by the way.

## Verify this pull request
This change added tests and can be verified as follows:
  - *Manually verified the change by testing locally.* 

Success to deploy and start the dolphinscheduler's images.
image

Visit the web ui and create a process named 'test':
image

Create a shell task named 'test':
image

And save and run the process:
image

### CODE REVIEW

File: docker/build/startup-init-conf.sh [hunk 1]
```diff
@@ -26,9 +26,8 @@ echo "init env variables"
 # Database
 #============================================================================
 [ "h2" == ${DATABASE_TYPE} ] || export DATABASE_TYPE=${DATABASE_TYPE:-"postgresql"}
-# export DATABASE_DRIVER=${DATABASE_DRIVER:-"org.postgresql.Driver"}
 [ "h2" == ${DATABASE_TYPE} ] || export DATABASE_HOST=${DATABASE_HOST:-"127.0.0.1"}
-[ "h2" == ${DATABASE_TYPE} ] || export DATABASE_PORT=${DATABASE_PORT:-"5432"}
+[ "h2" == ${DATABASE_TYPE} ] || export e=${DATABASE_PORT:-"5432"}
```
Comment (kezhenxu94) [inline docker/build/startup-init-conf.sh]: This is changed by mistake?

Comment (calvinjiang) [reply docker/build/startup-init-conf.sh]: I've committed it again.

**Comment by kezhenxu94** [comment]
Can you try the latest dev branch again, I've rework the Docker compose part recently and should fix these

**Comment by calvinjiang** [comment]
This PR has to be closed. Because the process of deploying docker has already changed in the branch of dev.

**Pull request closed without merging**